### ISSUE TITLE
Donate troops string recognition

### ISSUE THREAD

**Issue by Stormtrail** [CodeSlinger69/ClAsHbOt#14]
6/16/2015 11:04:17 AM DonateTroops()
 6/16/2015 11:04:18 AM Donate button found at absolute: 97, 152, 179, 177
 6/16/2015 11:04:18 AM Donate text: 'wizards or archers plz'
 6/16/2015 11:04:19 AM Donate troops window opened.
 6/16/2015 11:04:19 AM Donate loop 1 of 5
 6/16/2015 11:04:19 AM Troop DonateBarbarian.bmp found at 193, 76 conf: 99.92%
 6/16/2015 11:04:19 AM Troop DonateArcher.bmp found at 263, 77 conf: 99.93%
 6/16/2015 11:04:19 AM Troop DonateHogRider.bmp found at 470, 160 conf: 99.54%
 6/16/2015 11:04:19 AM String match for: Wizard, troop NOT available, exiting.

It appears to only recognize one match, but then if troop is not available, it never moves to the 2nd match. It also seems to be confused by requests like "arch wiz min" where all 3 are available donations but it is unable to donate.

**Comment by CodeSlinger69** [comment]
> It appears to only recognize one match, but then if troop is not available, it never moves to the 2nd match.

This is a "by design" kind of thing.  Please see this wiki page for an explanation: https://github.com/CodeSlinger69/ClAsHbOt/wiki/Auto-Donate

> It also seems to be confused by requests like "arch wiz min" where all 3 are available donations but it is unable to donate.

Yes, the donate code does not handle arbitrarily complex boolean logic.  I really didn't want to write a complete english language parser (and what about other languages?) to support complex requests. This is intentionally a very simple auto-donate system.  If you have ideas for how to implement something different that doesn't require a massive amount of complex code, I'm all ears!

**Comment by Stormtrail** [comment]
Code,

Sorry if these requests bothered you, seemed like I struck a nerve, this isn't intended as a whine or cry about the existing bot because the existing bot is great. I only filed them as separate issues because that seemed to be the way to have the discussions separated out. As I said before, I'm also looking for small little bits that I can learn from and adapt myself with your guidance.

I worked thru the config file and have it setup just fine for most of my purposes, it was just a failing of my understanding of the parsing in how it would implement them. I'm not that interested in mix donations or anything particularly complex, more looking for a way to functionally create a default donation.
